thunder/core/transform_common.py

```diff
@@ -363,6 +365,13 @@ def transform_state_dict_for_submodule(
         """
         return state_dict
 
+    def transform_state_dict(
```

Comment:
can we make a `reverse_transform_state_dict_for_submodule` instead, please?
(we also probably need the new parameters (`shared_names` and `processed_names` for shared params))